analysis_templates/01-single-group-integration-check-template.Rmd

```diff
@@ -508,6 +554,56 @@ if (celltypes_present) {
 ```
 
 
+### Within-batch ARI
+
+Here we calculate the what we will refer to as the within-batch ARI. 
+The metric measures how well a given integration method preserves biological heterogeneity by comparing pre- and post-intergration clustering assignments for each library. 
+This function takes each original SCE object, prior to any merging, and performs graph-based clustering. 
+The same clustering is then performed on the integrated object.
+For each library that is present in the integrated object, the original cluster assignments are compared to the cells' new cluster assignments post integration by calculating the ARI between these two cluster assignments. 
+Within-batch ARI values close to 1 indicate that biological heterogeneity has been preserved. 
+
+```{r}
+# map across the list of integrated sce objects to compare against the individual objects 
+within_batch_ari <- purrr::imap(sce_list, 
+                                \(sce, int_method){
+                                  calculate_within_batch_ari(individual_sce_list,
+                                                             sce,
+                                                             seed = params$seed,
+                                                             integration_method = int_method,
+                                                             batch_column = params$batch_column)
+                                }) |>
+  # combine results into one dataframe
+  dplyr::bind_rows()
+```
+
+```{r}
+# set order of integration methods on axes
+within_batch_ari <- set_integration_order(within_batch_ari)
+
+# plot within batch ARI by integration method
+ggplot(within_batch_ari, aes_string(x = "integration_method_factor" , y = "ari")) +
```

Comment:
> * for example:
> 
> ```
> a_real_function <- function(numeric_iris_column) {
>   ggplot(iris, aes(x = {{numeric_iris_column}})  + geom_histogram()
> }
> 
> # Usage is **without quotes**
> a_real_function(Sepal.Length)
> ```

I assume you meant something more like:

```
a_real_function <- function(data, numeric_column) {
  ggplot(data, aes(x = {{numeric_column}})  + geom_histogram()
}

a_real_function(iris, Sepal.Length)
```

Because I _know_ you would't use external variables in a function.

Reply:
I want everyone to know, and this is true, that I wrote that example intentionally without `iris` to see if it would elicit this exact comment, so this is an all-around team success 🎉 🎉 

Of course, I would _never_ do such a thing in production 😄